shm/map.hpp

```diff
@@ -1,17 +1,16 @@
 #pragma once
-#include <vector>
 #include <memory>
+#include <vector>
 #include "island.hpp"
 
-#define MAX_ISLANDS 10
 class map {
 public:
     map();
     void printIslands() const;
-    Island* getIsland( const Coordinates&)const;
+    Island* getIsland(const Coordinates&) const;
 
 private:
-    std::vector<std::pair<int, int>> generateIslandPosition();
-    std::vector<std::shared_ptr<Island>> map_;
-    Island* currentPosition_{};
+    std::vector<std::pair<int, int>> generateIslandPosition() const;
+    std::vector<std::shared_ptr<Island>> map_{};
+    Island* currentPosition_ = nullptr;
```

Comment:
Please be consistent. One time you initialize by `{}` other time by `=` use one style :)